src/IROperator.cpp

```diff
@@ -566,6 +568,26 @@ Expr lossless_cast(Type t, Expr e) {
         return Shuffle::make(vecs, shuf->indices);
     }
 
+    if (const Mod *mod = e.as<Mod>()) {
+        if (lossless_cast(t, mod->b).defined()) {
+            return cast(t, e);
+        } else {
+            return Expr();
+        }
+    }
+
+    if (const Ramp *ramp = e.as<Ramp>()) {
+        Type ty = t.with_lanes(ramp->type.lanes() / t.lanes());
+        Expr first = ramp->base;
+        Expr last = simplify(first + (ramp->lanes - 1) * ramp->stride);
```

Comment:
This seem correct but it's a little bit suspicious, and possibly is an issue to call simplify here. @abadams what do you think?

Reply:
The case that's dubious would be something like lossless_cast(UInt(8), ramp(u16(5), u16(32800), 3)). The first and last lanes are 5 and 69, which are both representable as UInt8s. But the middle lane is 32805, which is not.

It's correct for a type where overflow is not defined behavior, so you could only do this if it's a ramp of int32s, in case that helps.